publisher/publisher.go

```diff
@@ -55,20 +57,22 @@ type ctSubmissionRequest struct {
 
 // Impl defines a Publisher
 type Impl struct {
-	log          *blog.AuditLogger
-	client       *http.Client
-	issuerBundle []ct.ASN1Cert
-	ctLogs       []*Log
+	log               *blog.AuditLogger
+	client            *http.Client
+	issuerBundle      []ct.ASN1Cert
+	ctLogs            []*Log
+	submissionTimeout time.Duration
 
 	SA core.StorageAuthority
 }
 
 // New creates a Publisher that will submit certificates
 // to any CT logs configured in CTConfig
-func New(bundle []ct.ASN1Cert, logs []*Log) (pub Impl) {
+func New(bundle []ct.ASN1Cert, logs []*Log, submissionTimeout time.Duration) (pub Impl) {
 	logger := blog.GetAuditLogger()
 	logger.Notice("Publisher Authority Starting")
 
+	pub.submissionTimeout = submissionTimeout
```

Comment:
Let's take this opportunity change this to a more idiomatic style (returning a struct literal, not using named returns, and returning a pointer rather than a value):

```
return &Impl{
  submissionTimeout: submissionTimeout,
  ...
}
```